datumaro/plugins/data_formats/datumaro/exporter.py

```diff
@@ -376,13 +378,27 @@ class DatumaroExporter(Exporter):
     DEFAULT_IMAGE_EXT = DatumaroPath.IMAGE_EXT
     PATH_CLS = DatumaroPath
 
-    def create_writer(self, subset: str) -> _SubsetWriter:
+    def create_writer(
+        self, subset: str, images_dir: str, pcd_dir: str, related_images_dir: str
+    ) -> _SubsetWriter:
+        export_context = ExportContextComponent(
+            save_dir=self._save_dir,
+            save_media=self._save_media,
+            images_dir=images_dir,
+            pcd_dir=pcd_dir,
+            related_images_dir=related_images_dir,
+            crypter=NULL_CRYPTER,
+            image_ext=self._image_ext,
+            default_image_ext=self._default_image_ext,
+        )
+
         return _SubsetWriter(
             context=self,
             ann_file=osp.join(self._annotations_dir, subset + self.PATH_CLS.ANNOTATION_EXT),
+            export_context=export_context,
         )
 
-    def apply(self):
+    def apply(self, pool: Optional[Pool] = None):
```

Comment:
How about just accepting all arguments since `_SubsetWriter` could parents of other than one from `datumaro_binary`?
```suggestion
    def apply(self, *args, **kwargs):
```

Reply:
Same here.